pyxem/utils/ransac_ellipse_tools.py

```diff
@@ -574,3 +576,118 @@ def _get_ellipse_markers(
+                      **kwargs,
+                      ):
+    """
+    This method starts by taking some number of points which are the most intense
+    in the signal.  It then takes those points and guesses some starting parameters
+    for the `get_ellipse_model_ransac_single_frame` function. From there it will try
+    to determine the ellipse parameters.
+
+    Parameters
+    -----------
+    signal: Signal2D
+        The signal of interest
+    mask: Array-like
+        The mask to be applied to the data.  All of the masked values are ignored
+    num_points: int
+        The number of points to consider
+    guess_starting_params: bool
+        If True then the starting parameters will be guessed based on the points determined.
+    **kwargs:
+        Any other keywords for ` get_ellipse_model_ransac_single_frame`
+
+    Returns
+    -------
+    center: (x,y)
+        The center of the diffraction pattern
+    affine:
+        The affine transformation to make the diffraction pattern circular.
+    """
+    pos = get_max_positions(signal,
+                            mask=mask,
+                            num_points=num_points)
+    import matplotlib.pyplot as plt
+    plt.scatter(pos[:, 0], pos[:, 1])
+    if use_ransac:
+        if guess_starting_params:
+            el, _ = get_ellipse_model_ransac_single_frame(pos,
+                                                  xf=np.mean(pos[:, 0]),
+                                                  yf=np.mean(pos[:, 1]),
+                                                  rf_lim=np.shape(signal.data)[0]/5,
+                                                  semi_len_min=np.std(pos[:, 1]),
+                                                  semi_len_max=np.std(pos[:, 1])*2,
+                                                  semi_len_ratio_lim=1.2,
+                                                  min_samples=6,
+                                                  residual_threshold=20,
+                                                  max_trails=1000)
+        else:
+            el, _ = get_ellipse_model_ransac_single_frame(pos,
+                                                        **kwargs)
+    else:
+        e = EllipseModel()
+        converge = e.estimate(data=pos)
+        el = e
+    if el is not None:
+        affine = ellipse_to_affine(el.params[3],el.params[2], el.params[4])
+        center = (el.params[0],el.params[1])
+        return center, affine
+    else:
+        print("Ransac Ellipse detection did not converge")
+        return None
+
+
+def ellipse_to_affine(major, minor, rot):
+    if major < minor:
+        print("major<minor")
+        temp = major
+        major = minor
+        minor = temp
+        rot=rot+(np.pi/2)
+
+    Q = [[np.cos(rot), -np.sin(rot), 0],
+         [np.sin(rot), np.cos(rot), 0],
+         [0, 0, 1]]
+    S = [[1, 0, 0],
+         [0, major / minor, 0],
+         [0, 0, 1]]
+    C = np.matmul(np.matmul(Q, S), np.transpose(Q))
```

Comment:
Yea it just assumes everything is the same.  It could be made slightly better and you could use it for center the zero beam if there is a beam stop.  That would require summing up individual parts of the pattern however